server/server.go

```diff
@@ -0,0 +1,129 @@
+// Package server provides a server implementation to connect network transport
+// protocols and service business logic by defining server endpoints.
+package server
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"github.com/giantswarm/microerror"
+	microserver "github.com/giantswarm/microkit/server"
+	"github.com/giantswarm/micrologger"
+	kithttp "github.com/go-kit/kit/transport/http"
+
+	"github.com/giantswarm/node-operator/server/endpoint"
+	"github.com/giantswarm/node-operator/server/middleware"
+	"github.com/giantswarm/node-operator/service"
+)
+
+// Config represents the configuration used to create a new server object.
+type Config struct {
+	// Dependencies.
+	Service *service.Service
+
+	// Settings.
+	MicroServerConfig microserver.Config
+}
+
+// DefaultConfig provides a default configuration to create a new server object
+// by best effort.
+func DefaultConfig() Config {
```

Comment:
Maybe remove DefaultConfig() like elsewhere.

Reply:
I am not doing this here now because there seems to be more to it. The default config is not trivial and I want to focus on the matter of the node-operator now. We have to boyscout these things in other projects before going full into it in the operators.